types/set.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	"sync"
+)
+
+type Set struct {
```

Comment:
Wasn't the point of #387 to phase out these structures?

Reply:
See above